### PR TITLE
api(setInputFiles): introduce page/frame helpers, document, break compat

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -3106,6 +3148,55 @@ Returns path to the downloaded file in case of successful download.
 Returns downloaded url.
 
 
+### class: FileChooser
+
+[FileChooser] objects are dispatched by the page in the ['filechooser'](#event-filechooser) event.
+
+```js
+page.on('filechooser', fileChooser => {
```
Comment (dgozman) [inline docs/api.md:3172]: async fileChooser =>

File: docs/api.md [hunk 2]
```diff
@@ -757,15 +759,13 @@ Emitted when attachment download started. User can access basic file operations
...
+- <[FileChooser]>
 
-Emitted when a file chooser is supposed to appear, such as after clicking the  `<input type=file>`. Playwright can respond to it via setting the input files using [`elementHandle.setInputFiles`](#elementhandlesetinputfilesfiles-options) which can be uploaded in the end.
+Emitted when a file chooser is supposed to appear, such as after clicking the  `<input type=file>`. Playwright can respond to it via setting the input files using [`fileChooser.setFiles`](#filechoosersetfilesfiles-options) that can be uploaded after that.
 
 ```js
-page.on('filechooser', async ({element, multiple}) => {
-  await element.setInputFiles('/tmp/myfile.pdf');
+page.on('filechooser', fileChooser => {
+  await fileChooser.setFiles('/tmp/myfile.pdf');
```
Comment (dgozman) [inline docs/api.md:768]: await requires an async function

File: docs/uploadDownload.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+## Upload a file
+
+```js
+// <input id=upload type=file>
+
+await page.setInputFiles('input#upload', 'myfile.pdf');
+```
+
+You can select input files for upload using the `page.setInputFiles` method. It expects first arcument to point to an [input element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) with the type `"file"`. Multiple files can be passed in the array. If some of the file paths are relative, they are resolved relative to the [current working directory](https://nodejs.org/api/process.html#process_process_cwd). Empty array clears the selected files.
```
Comment (dgozman) [inline docs/uploadDownload.md:11]: typo: arcument

File: docs/uploadDownload.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+Sometimes element that picks files appears dynamically. When this happens, [`"filechooser"`](https://github.com/microsoft/playwright/blob/master/docs/api.md#event-filechooser) event is emitted on the page. It contains the [`FileChooser`](https://github.com/microsoft/playwright/blob/master/docs/api.md#class-filechooser) object that can be used to select files:
+
+```js
+const [ fileChooser ] = await Promise.all([
+	page.waitForEvent('filechooser'), // <-- start waiting for the file chooser
+	page.click('button#delayed-select-files') // <-- perform the action that directly or indirectly initiates it.
+]);
+// Now that both operations resolved, we can use the returned value to select files.
+fileChooser.setFiles(['file1.txt', 'file2.txt'])
```
Comment (dgozman) [inline docs/uploadDownload.md:49]: await?

File: src/injected/injected.ts [hunk 5]
```diff
@@ -248,6 +248,29 @@ class Injected {
     throw new Error('Not a checkbox');
   }
 
+  async setFileInputFunction(node: Node, payloads: types.FileTransferPayload[]) {
```
Comment (dgozman) [inline src/injected/injected.ts:251]: nit: just setInputFiles?

**Review by dgozman — APPROVED** [review]

File: docs/api.md (hunk 1 above)
Comment (pavelfeldman) [reply docs/api.md:3172]: Done

File: docs/api.md (hunk 2 above)
Comment (pavelfeldman) [reply docs/api.md:768]: Done

File: docs/uploadDownload.md (hunk 3 above)
Comment (pavelfeldman) [reply docs/uploadDownload.md:11]: Done

File: src/injected/injected.ts (hunk 5 above)
Comment (pavelfeldman) [reply src/injected/injected.ts:251]: Done

File: docs/uploadDownload.md (hunk 4 above)
Comment (pavelfeldman) [reply docs/uploadDownload.md]: Done

**Pull request merged**